sample/marchingSquares.glsl

```diff
@@ -0,0 +1,204 @@
+#include "../generative/pnoise.glsl"
+#include "../math/const.glsl"
+#include "../math/rotate2d.glsl"
+#include "../sample.glsl"
+
+/*
+Author: Guido Schmidt
+description: The Marching Squares algorithm generates contour lines (isolines) of a given input for a
+given grid and a threshhold. Is uses a lookup of 16 cases to decide if a cell is inside, outside or
+a specific corner.
+use: <vec3> marchingSquares(in <vec2> uv) 
+options:
+    - SAMPLE_MARCHING_SQUARES_FNC(UV): optional sampling function, defaults so sampling of pnoise
+*/
+
+#ifndef SAMPLE_MARCHING_SQUARES_FNC
+float defaultSampleFunction(in vec2 uv) {
+    float dv = pnoise(uv * 5.0, vec2(0.0));
+    return dv;
+}
+#define SAMPLE_MARCHING_SQUARES_FNC(UV) defaultSampleFunction(UV)
+#endif
+
```

Comment:
We use the following pattern to prevent name collisions

```
#ifndef FNC_SAMPLEMARCHINGSQUARES 
#define FNC_SAMPLEMARCHINGSQUARES
...
#endif
```

Reply:
Sorry this comment I made was a bit confusing. 

```
#ifndef FNC_SAMPLEMARCHINGSQUARES 
#define FNC_SAMPLEMARCHINGSQUARES
...
#endif
```

is to wrap the entire function. For example here: https://github.com/patriciogonzalezvivo/lygia/blob/main/sample/shadowPCF.glsl#L18-L19

While for "templeting" just the function use to do sampling have this other pattern:

```
#ifdef SAMPLEMARCHINGSQUARES_SAMPLE_FNC
#define SAMPLEMARCHINGSQUARES_SAMPLE_FNC(TEX, UV) defaultSampleFunction(TEX, UV)
#endif
```

For example:

Reply:
Ah yes, got it. Really like in C++ world with the header files, thanks for clarifying